// python -m grpc_tools.protoc -I../proto --python_out=. --pyi_out=. --grpc_python_out=. matcher.proto

syntax = "proto3";

package sk.stuba.fei.api.msus.dp.matcher;

option java_multiple_files = true;

service Matcher {
  rpc MatchFingerprints(MatchRequest) returns (MatchResponse) {}
  rpc MatchIrises(MatchRequest) returns (MatchResponse) {}
}

message MatchRequest {
  MatchTemplate sampleTemplate = 1;
  MatchTemplate realTemplate = 2;
}

message MatchResponse {
  float matchScore = 1;
}

message MatchTemplate {
  int32 keypointsSize = 1;
  string encodedDescriptor = 2;
}
